File: packages/server/src/__tests__/fixtures/python/Dog.py
"""
Dog class for testing Python parser with inheritance
"""

from Animal import Animal


class Dog(Animal):
    """Dog class that extends Animal"""
    
    def __init__(self, name: str, age: int, breed: str):
        """Initialize dog with name, age, and breed"""
        super().__init__(name, age)
        self.breed = breed
    
    def speak(self) -> str:
        """Dog makes a woof sound"""
        return "Woof"
    
    def get_breed(self) -> str:
        """Get dog breed"""
        return self.breed

File: packages/server/src/__tests__/fixtures/java/Dog.java
package com.example;

/**
 * Dog class for testing Java parser with inheritance
 */
public class Dog extends Animal {
    private String breed;
    
    public Dog(String name, int age, String breed) {
        super(name, age);
        this.breed = breed;
    }
    
    @Override
    public String speak() {
        return "Woof";
    }
    
    public String getBreed() {
        return breed;
    }
}
